[Bug]: Conflicting peer dependency for glob after updating to 7.0.0-beta-42

### Describe the bug

We receive the following warnings after updating to `7.0.0-beta-42`:

```bash
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: glob-promise@4.2.2
npm WARN Found: glob@8.1.0
npm WARN node_modules/glob
npm WARN   dev glob@"8.1.0" from the root project
npm WARN   6 more (@npmcli/map-workspaces, cacache, cacache, ng-packagr, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer glob@"^7.1.6" from glob-promise@4.2.2
npm WARN node_modules/glob-promise
npm WARN   glob-promise@"^4.2.0" from @storybook/core-common@7.0.0-beta.43
npm WARN   node_modules/@storybook/core-common
npm WARN
npm WARN Conflicting peer dependency: glob@7.2.3
npm WARN node_modules/glob
npm WARN   peer glob@"^7.1.6" from glob-promise@4.2.2
npm WARN   node_modules/glob-promise
npm WARN     glob-promise@"^4.2.0" from @storybook/core-common@7.0.0-beta.43
npm WARN     node_modules/@storybook/core-common
```


### To Reproduce

_No response_

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.22621
    CPU: (24) x64 AMD Ryzen 9 3900X 12-Core Processor
  Binaries:
    Node: 16.16.0 - C:\dev\nodejs\node.EXE
    npm: 8.13.2 - C:\dev\nodejs\npm.CMD
  Browsers:
    Chrome: 109.0.5414.120
    Edge: Spartan (44.22621.1105.0), Chromium (108.0.1462.76)
  npmPackages:
    @storybook/addon-actions: 7.0.0-beta.43 => 7.0.0-beta.43
    @storybook/addon-essentials: 7.0.0-beta.43 => 7.0.0-beta.43
    @storybook/addon-links: 7.0.0-beta.43 => 7.0.0-beta.43
    @storybook/addon-postcss: 2.0.0 => 2.0.0
    @storybook/addons: 7.0.0-beta.43 => 7.0.0-beta.43
    @storybook/angular: 7.0.0-beta.43 => 7.0.0-beta.43
    @storybook/builder-webpack5: 7.0.0-beta.43 => 7.0.0-beta.43
    @storybook/core-server: 7.0.0-beta.43 => 7.0.0-beta.43
```
```


### Additional context

Looks like this was caused by: https://github.com/storybookjs/storybook/pull/20927

`glob-promise` may need to be updated as well here: https://github.com/storybookjs/storybook/blob/next/code/lib/core-common/package.json#L62

`glob-promise@5` is the minimum version that allows `glob@8`: https://github.com/ahmadnassri/node-glob-promise/blob/v5.0.0/package.json#L42

--- a/code/lib/builder-vite/package.json
+++ b/code/lib/builder-vite/package.json
@@ -58,7 +58,7 @@
     "express": "^4.17.3",
     "fs-extra": "^11.1.0",
     "glob": "^8.1.0",
-    "glob-promise": "^4.2.0",
+    "glob-promise": "^6.0.2",
     "magic-string": "^0.27.0",
     "rollup": "^2.25.0 || ^3.3.0",
     "slash": "^3.0.0"

--- a/code/lib/core-common/package.json
+++ b/code/lib/core-common/package.json
@@ -59,7 +59,7 @@
     "find-up": "^5.0.0",
     "fs-extra": "^11.1.0",
     "glob": "^8.1.0",
-    "glob-promise": "^4.2.0",
+    "glob-promise": "^6.0.2",
     "handlebars": "^4.7.7",
     "lazy-universal-dotenv": "^4.0.0",
     "picomatch": "^2.3.0",

